<a href="https://colab.research.google.com/github/c212/a310-fall2026-yaljabri/blob/main/Warmup_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Problem 1

A fair game has an expected value of 0. The probability of rolling a 7 is \(6/36 = 1/6\), and the probability of losing is \(5/6\).

\[
E = 4(1/6) - 1(5/6) = -1/6
\]

Therefore, the game is not fair because the player loses an average of \(1/6\) of a chip per game. A fair prize would be 5 chips plus the return of the original chip.

In [4]:
import random

times = 1_000_000
money = 0

for i in range(times):
    die1 = random.randint(1, 6)
    die2 = random.randint(1, 6)

    if die1 + die2 == 7:
        money += 4
    else:
        money -= 1

print(money / times)

-0.16719


The simulation gives a result close to \(-0.167\), which means the player loses about 0.167 chips per game. Therefore, the game is not fair.

Problem 2

In [5]:
count = 0

for d1 in range(1, 7):
    for d2 in range(1, 7):
        for d3 in range(1, 7):
            dice_sum = d1 + d2 + d3

            if dice_sum not in [3, 5, 7, 11, 13, 17]:
                count += 1

print(count / (6 * 6 * 6))

0.6620370370370371


Out of 216 possible outcomes, 143 have a non-prime sum. Therefore, the probability is \(143/216\), or approximately 0.662 (66.2%).

In [6]:
import random

times = 1_000_000
count = 0

for i in range(times):
    d1 = random.randint(1, 6)
    d2 = random.randint(1, 6)
    d3 = random.randint(1, 6)

    dice_sum = d1 + d2 + d3

    if dice_sum not in [3, 5, 7, 11, 13, 17]:
        count += 1

print(count / times)

0.660993


The simulation gives a result close to 0.662, meaning there is approximately a 66.2% chance that the sum is not prime. This agrees with the exact result.

Problem 3

In [7]:
lockers = [False] * 100

for student in range(1, 101):
    for locker in range(student - 1, 100, student):
        lockers[locker] = not lockers[locker]

    stage = ""

    for locker in lockers:
        if locker:
            stage += "O"
        else:
            stage += "X"

    print("After student", student, ":", stage)

print("\nOpen lockers:")

for i in range(100):
    if lockers[i]:
        print(i + 1, end=" ")

After student 1 : OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
After student 2 : OXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOXOX
After student 3 : OXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXXOOOXXX
After student 4 : OXXOOOOOXXOXOXXOOOOOXXOXOXXOOOOOXXOXOXXOOOOOXXOXOXXOOOOOXXOXOXXOOOOOXXOXOXXOOOOOXXOXOXXOOOOOXXOXOXXO
After student 5 : OXXOXOOOXOOXOXOOOOOXXXOXXXXOOXOOXXXXOXXXOOOOOXOXOOXOOOXOXXOOOXXOXOOOXOOXOXOOOOOXXXOXXXXOOXOOXXXXOXXX
After student 6 : OXXOXXOOXOOOOXOOOXOXXXOOXXXOOOOOXXXOOXXXOXOOOXOOOOXOOXXOXXOXOXXOXXOOXOOOOXOOOXOXXXOOXXXOOOOOXXXOOXXX
After student 7 : OXXOXXXOXOOOOOOOOXOXOXOOXXXXOOOOXXOOOXXXOOOOOXOOXOXOOXXXXXOXOXOOXXOOXXOOOXOOXXOXXXOXXXXOOOXOXXXOOOXX
After student 8 : OXXOXXXXXOOOOOOXOXOXOXOXXXXXOOOXXXOOOXXOOOOOOXOXXOXOOXXOXXOXOXOXXXOOXXOXOXOOXXOOXXOXXXXXOOXOXXXXOOXX
After student 9 : OXXOXXXXOOOOOOOXOOOXOXOXXXOXOO

Each student changes every locker whose number is a multiple of the student's number. A locker is changed once for every factor it has.

Most numbers have factors in pairs, so their lockers are changed an even number of times and finish closed. Perfect-square numbers have one unpaired factor, so they are changed an odd number of times and remain open.

Therefore, the open lockers are:

\(1, 4, 9, 16, 25, 36, 49, 64, 81,\) and \(100\).